src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java

```diff
@@ -228,5 +260,39 @@ public String getUntranslatedHost() {
     return "";
   }
 
+  private Job getJobStatus(String id) {
+    // Poll with HTTP client the job's status
+    return new Job("repair", "repair-12345678"); // Stub, but this is the format that mgmt-api returns
+  }
+
+  private void scheduleJobPoller(ScheduledExecutorService scheduler, int pollInterval) {
```

Comment:
Question: I'm confused by how the concurrency is meant to work here. Intuitively, I'd have thought that scheduleJobPoller should be called for each repairID and should then add an executor to `repairStatusExecutors` to commence polling the endpoint . But it seems that instead `triggerRepair` is called for each repairID, and adds to both `repairStatusExecutors` and `repairStatusHandlers`, but not to `jobTracker`.

So a few questions about this:

1. Why do we need a separate `repairStatusHandler` for each repair? Is that the way the current JMX implementation works? It looks like maybe, given that new `SegmentRunners` (the implementor of `RepairStatusHandler` are created for every call of `repairSegment` and presumably this is called per segment).
2. Maybe if you could let me know what each of the below maps is meant to hold this would help me understand where you're going with this:

```
private final ConcurrentMap<Integer, RepairStatusHandler> repairStatusHandlers = Maps.newConcurrentMap();
  private final ConcurrentMap<String, JobStatusTracker> jobTracker = Maps.newConcurrentMap();
  private final ConcurrentMap<Integer, ExecutorService> repairStatusExecutors = Maps.newConcurrentMap();
```

Reply:
> Why do we need a separate repairStatusHandler for each repair? Is that the way the current JMX implementation works?

Yes, this is how the JMX implementation works.

> private final ConcurrentMap<Integer, RepairStatusHandler> repairStatusHandlers = Maps.newConcurrentMap();
 > private final ConcurrentMap<Integer, ExecutorService> repairStatusExecutors = Maps.newConcurrentMap();

These are both the same as in JMX implementation, since the implementation itself is pretty much the same as the JMX one. No logic changes. 

> private final ConcurrentMap<String, JobStatusTracker> jobTracker = Maps.newConcurrentMap();

This is used to track the last seen notification from the mgmt-api side. So that we know which notifications are to be sent to the repairStatusHandlers and which ones have been processed already.